File: python/1351_Count_Negative_Numbers_in_a_Sorted_Matrix.py
class Solution:
    def countNegatives(self, grid) -> int:
        l = len(grid[0])
        j = l - 1
        ans = 0

        for i in range(len(grid)):
            while (j >= 0 and grid[i][j] < 0):
                j -= 1
            ans += (l - j - 1)
        
        return ans

File: cpp/1351_Count_Negative_Numbers_in_a_Sorted_Matrix.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int l = grid[0].size();
        int j = l - 1;
        int ans = 0;

        for (int i = 0; i < grid.size(); i++) {
            while (j >= 0 && grid[i][j] < 0) {
                j--;
            }
            ans += (l - j - 1);
        } 

        return ans;
    }
};